src/mlpack/methods/ann/init_rules/orthogonal_init.hpp

```diff
@@ -38,13 +38,14 @@ class OrthogonalInitialization
    * @param rows Number of rows.
    * @param cols Number of columns.
    */
-  template<typename eT>
-  void Initialize(arma::Mat<eT>& W, const size_t rows, const size_t cols)
+  template<typename MatType>
+  void Initialize(MatType& W, const size_t rows, const size_t cols)
   {
-    arma::Mat<eT> V;
-    arma::Col<eT> s;
+    MatType V;
+    typedef typename GetColType<MatType>::type ColType;
+    ColType s;
 
-    arma::svd_econ(W, s, V, arma::randu<arma::Mat<eT> >(rows, cols));
+    svd_econ(W, s, V, arma::randu<MatType>(rows, cols));
```

Comment:
Note that Bandicoot doesn't currently have `svd_econ()`, but we can fix that later of course. :+1:

Reply:
Yes, all of these need to be handled later, I will add this to the list in bandicoot we will do it in the next version.
I do not think a lot of us are using orthogonal initialisation for now